src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -0,0 +1,90 @@
+import { parse } from 'url';
+import type { Provider } from 'oidc-provider';
+// This import probably looks very hacky and it is. Weak Cache is required to get the oidc
+// configuration, which, in turn, is needed to get the routes the provider is using.
+// It is probably very difficult to get the configuration because Panva does not want
+// it to be possible, but we must get the configuration to satisfy the needs of the CSS
+// architecture. See the "canHandle" method for an explantaion
+// eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import instance from 'oidc-provider/lib/helpers/weak_cache';
+import type { HttpHandlerInput } from '../server/HttpHandler';
+import { HttpHandler } from '../server/HttpHandler';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { IdentityProviderFactory } from './IdentityProviderFactory';
+import type { IdPInteractionPolicyHttpHandler } from './interaction/IdPInteractionPolicyHttpHandler';
+
+export class IdentityProviderHttpHandler extends HttpHandler {
+  private readonly provider: Provider;
+  private readonly interactionPolicyHttpHandler: IdPInteractionPolicyHttpHandler;
+
+  public constructor(
+    providerFactory: IdentityProviderFactory,
+    interactionPolicyHttpHandler: IdPInteractionPolicyHttpHandler,
+  ) {
+    super();
+    this.interactionPolicyHttpHandler = interactionPolicyHttpHandler;
+    this.provider = providerFactory.createProvider(this.interactionPolicyHttpHandler);
+  }
+
+  /**
+   * Handles a request. Returns a promise that will either resolve if a response is
+   * given (including if the response is an error page) and throw an error if the
+   * idp cannot handle the request.
+   * NOTE: This method has a lot of hacks in it to get it to work with node-oidc-provider.
+   */
```

Comment:
At the moment, I don't understand what the method is doing.

Can we add that as a comment, so that we can look for a solution to not have it be a hack?

I think we just need to map the Panva router to a subpath or so, such that we can easily check in `canHandle` whether or not we are on the subpath.

Reply:
I've rewritten the description

Add what as a comment?

We could map it to a subpath, but even then, IdentityProviderHttpHandler is designed to operate not knowing that it's attached to a subpath. It's agnostic, so we still need to perform checks to see if it can handle the request.